OpenRA.Game/Game.cs

```diff
@@ -891,14 +891,30 @@ public static void AddSystemLine(string text)
 			AddSystemLine(systemMessageLabel, text);
 		}
 
-		public static void AddSystemLine(string name, string text)
+		public static void AddSystemLine(string prefix, string text)
 		{
-			OrderManager.AddChatLine(name, systemMessageColor, text, systemMessageColor);
+			AddChatLine(prefix, systemMessageColor, text, systemMessageColor, ChatPool.System);
 		}
 
 		public static void AddChatLine(string name, Color nameColor, string text)
 		{
-			OrderManager.AddChatLine(name, nameColor, text, chatMessageColor);
+			AddChatLine(name, nameColor, text, chatMessageColor, ChatPool.Chat);
+		}
+
+		public static void AddChatLine(string name, Color nameColor, string text, Color textColor,  ChatPool pool)
```

Comment:
Also IMO we now have way too many (7!) different methods for adding chat lines. How disruptive would it be to cut this down to require all callers to specify their pool (and IMO move that to the first argument).

Maybe something like this?
```csharp
public static void AddChatLine(ChatPool pool, string text, Color? textColor = null);
public static void AddChatLine(ChatPool pool, string prefix, string text, Color? prefixColor = null, Color? textColor = null);
```
Don't force things into these specific signatures if these are not the most natural for our use cases.

Shouldn't we also be doing the `ChatPoolFilters` check here? Code that calls this directly with `ChatPool.Feedback` will ignore filtering.

Reply:
Yes, I also think there are too many methods. I talked about this in the previous PR (see https://github.com/OpenRA/OpenRA/pull/17737#discussion_r460856381). I like the idea of requiring callers to specify a pool :+1: I will try to improve this.

Reply:
Ugh, this needs more work but I definitely want to base it on #19360

Reply:
I went over this a few times and after all I like it better with separate methods than requiring a pool to be specified each time. Having several methods is less of a problem now that chat handling is no longer cluttering `Game.cs`. I changed the method signature a bit after your suggestion settling on this:
```csharp
public static void AddChatLine(string prefix, string text, Color? prefixColor = null, Color? textColor = null, TextNotificationPool? pool = null);
```